packages/vite/src/node/certificate.ts

```diff
@@ -0,0 +1,145 @@
+// simplified fork of
+// https://github.com/jfromaniello/selfsigned/blob/da38146f8d02183c35f49f91659a744a243e8707/index.js
+// with inlined options and partial node-forge usage
+// to achieve smaller bundle
+//
+// this utility create untrusted certificate which still
+// allows to access page after proceeding a wall with warning
+//
+// should be deprecated eventually and replaced with recipes
+// about generating secure trusted certificates
+
+// @ts-ignore
+import forge from 'node-forge/lib/forge'
+// @ts-ignore
+import 'node-forge/lib/pki'
+
+// a hexString is considered negative if it's most significant bit is 1
+// because serial numbers use ones' complement notation
+// this RFC in section 4.1.2.2 requires serial numbers to be positive
+// http://www.ietf.org/rfc/rfc5280.txt
+function toPositiveHex(hexString: string) {
+  let mostSiginficativeHexAsInt = parseInt(hexString[0], 16)
+  if (mostSiginficativeHexAsInt < 8) {
+    return hexString
+  }
+
+  mostSiginficativeHexAsInt -= 8
+  return mostSiginficativeHexAsInt.toString() + hexString.substring(1)
+}
+
+export function createCertificate(): string {
+  const days = 30
```

Comment:
add more days, the limit is too short, I configure the ssl/tls certificates with 2 years

Reply:
This is what we have with existing logic. I think it's ok to keep as is if we are gonna deprecate and remove this feature eventually.

Reply:
@TrySound can you provide a certificate generated with the code here?, my email is userquin@gmail.com